doc/applications.md

```diff
@@ -0,0 +1,151 @@
+const state$ = increase$.scan((state, changeFn) => changeFn(state), {count: 0});
+```
+
+We can now add a a couple of more observables which will also change the same state store.
+
+```js
+const increaseButton = document.querySelector('#increase');
+const increase$ = Rx.Observable.fromEvent(increaseButton, 'click')
+  .map(() => state => {...state, count: state.count + 1});
+
+const decreaseButton = document.querySelector('#decrease');
+const decrease$ = Rx.Observable.fromEvent(decreaseButton, 'click')
+  .map(() => state => {...state, count: state.count - 1});
+
+const input = document.querySelector('#input');
+const input$ = Rx.Observable.fromEvent(input, 'keypress')
+  .map(event => state => {...state, inputValue: event.target.value});
+
+// We merge the three state change producing observables
+const state$ = Rx.Observable.merge(
+  increase$,
+  decrease$,
+  input$
+).scan((state, changeFn) => changeFn(state), {
+  count: 0,
+  inputValue: ''
+});
+
+// We subscribe to state changes and update the DOM
+state$.subscribe((state) => {
+  document.querySelector('#count').innerHTML = state.count;
+  document.querySelector('#hello').innerHTML = 'Hello ' + state.inputValue;
+});
+```
+
+We can take the state store approach and use it with many different frameworks and libraries.
+
+### Immutable JS
+You can also create a global state store for your application using [Immutable JS](https://facebook.github.io/immutable-js/). Immutable JS is a great way to create immutable state stores that allows you to optimize rendering by doing shallow checks on changed values.
+
+```js
+import Immutable from 'immutable';
+import someObservable$ from './someObservable';
+import someOtherObservable$ from './someOtherObservable';
+
+const initialState = {
+  foo: 'bar'
+};
+
+const state$ = Observable.merge(
+  someObservable$,
+  someOtherObservable$
+).scan((state, changeFn) => changeFn(state), Immutable.fromJS(initialState));
+
+export default state$;
+```
+
+Now you can import your state in whatever UI layer you are using.
+
+```js
+import state$ from './state';
+
+state$.subscribe(state => {
+  document.querySelector('#text').innerHTML = state.get('foo');
+});
+```
+
+### React
+To reduce boilerplate you can use [rxjs-react-component](https://www.npmjs.com/package/rxjs-react-component). It will allow you to expose observables that maps to state changes, as if you would do `this.setState({count: 1})`.
+
+```js
+class MyComponent extends ObservableComponent {
+  constructor(props) {
+    super(props);
+    this.state = {count: 0};
+  }
+  getObservables() {
+    return {
+      onClick$(observable, getState) {
+        return observable.map(() => ({count: getState().count + 1}));
+      }
+    };
+  }
+  render() {
+    return (
+      <div>
+        <div>{this.state.count}</div>
+        <button onClick={this.observables.onClick$}>Increase count</button>
+      </div>
+    );
+  }
+}
+
+export default MyComponent;
+```
+
+### Angular2
+Coming soon
```

Comment:
I can try to get someone from the Angular team to help with this. cc/ @jeffbcross @robwormald @vsavkin @bradlygreen